graphbolt/src/cuda/index_select_impl.cu

```diff
@@ -46,19 +107,41 @@ torch::Tensor UVAIndexSelectImpl_(torch::Tensor input, torch::Tensor index) {
                                       .dtype(input.dtype())
                                       .device(c10::DeviceType::CUDA));
   DType* input_ptr = input.data_ptr<DType>();
-  IdType* index_ptr = index.data_ptr<IdType>();
   DType* ret_ptr = ret.data_ptr<DType>();
+
+  // Sort the index to improve the memory access pattern.
+  torch::Tensor sorted_index, permutation;
+  std::tie(sorted_index, permutation) = torch::sort(index);
+  const IdType* index_sorted_ptr = sorted_index.data_ptr<IdType>();
+  const int64_t* permutation_ptr = permutation.data_ptr<int64_t>();
+
   cudaStream_t stream = 0;
-  dim3 block(512, 1);
-  // Find the smallest block size that can fit the feature_size.
-  while (static_cast<int64_t>(block.x) >= 2 * feature_size) {
-    block.x >>= 1;
-    block.y <<= 1;
+
+  if (feature_size == 1) {
```

Comment:
Just to confirm, it the feature_size here the size of feature for each node or each edge?
Can you add documentation at the top of the cu file to explain how each option here improves each situation?

Reply:
> is the feature_size here the size of feature for each node or each edge?

Can be either node or edge feature. The index_select operator is designed for our feature store.